### ISSUE TITLE
9.4.2 seems to have introduced a "No UPnP device with public ip was found" error.

### ISSUE THREAD

**Issue by Ohnix-Ahnul** [StrangeLoopGames/EcoIssues#22694]
I am experiencing frequent drops with the new 9.4.2 patch.  Examination of the log file(s) show a consistent cause for the drops "No UPnP device with public ip was found".  I am running as single player with the base ECOserver executable and the game is a few weeks old and I have a few MODs installed that were working well with 9.4.1.  There seem to be frequent periods of where the game seems to hang for a 0-2 seconds or so.  Then after a few minutes the game drops me and I find that the latest log shows the '...No UPnP...' error as the cause of the drop.

Hate to complicate things but, ... , I had ECO Server setup but did not find a use for it and then removed the folder and files from my PC.  This was before starting the current game.  And when I set that one up I slightly modified a couple of the base config files notably WorldGenerator.eco.  But I did not uninstall ECO Server from the PC until I started getting these "No UPnP device with public ip was found" drops.  After I uninstalled I am still able to launch ECO through steam (shortcut on desktop) and begin to notice the periods of no response from the game are more frequent and sometime soon thereafter the game drops me again.

attached latest log file:
[log_210922103037.log](https://github.com/StrangeLoopGames/EcoIssues/files/7215366/log_210922103037.log)

**Comment by Ohnix-Ahnul** [comment]
Managed to find a way to disable the UPdP server setting in the ECOServer 0.9.4.2 beta release-256 GUI {launched from EcoServer.exe).   But this did not prevent the game from shutting down.  This time the log file shows "Parent process exited, shutting down" about 18 minutes after starting the game through Steam.  And this is not really helpful as there is no indication of what caused it.
I have the ECOServer.exe running again to see if it will generate a similar error and drop the game after 15-30 minutes and it's at the 15m mark right now.  And the game is not running on Steam so there is nothing to interact with besides the Admin GUI.  So I'll let it run for another 30-60 minutes before calling it a night and seeing if anything can be done tomorrow.

ah, log file of last Steam run:
[log_210923120852.log](https://github.com/StrangeLoopGames/EcoIssues/files/7215810/log_210923120852.log)

ok, ECOServer.exe and the Admin GUI have been running for about 1h and 23m with no termination.   I'll try more ECO runs on Steam tomorrow and see how long they manage to stay up before being dropped.

oh, log of the latest ECOServer.exe run (that didn't get dropped):
[log_210923123201.log](https://github.com/StrangeLoopGames/EcoIssues/files/7216177/log_210923123201.log)

**Comment by mirasrael** [comment]
> [00:27:15.464] [96] [ Info] [Eco] Parent process exited, shutting down.
It usually happen when started from client and the client exited. 
You can ignore UPnP message. It is just warning about improper setup, but won't affect how server running.

**Comment by Ohnix-Ahnul** [comment]
Thanks for the reply Mirasrael.  It got me looking more at the mods and I found that a lot of the EM mods I'm using have been updated yesterday/today.  After updating the mods I started up ECO as usual and had it running for about 9 and 1/2 hours without issue.  So it appears to be working smoothly again.  In the future, before posting, I'll check to see if other similar issues have been reported and then wait on updates before posting.  Or just post if the issue(s) appears to be unique.
